### DISCUSSION TITLE
Does anyone here know why math functions are not grouped under some Math class in GDScript?

### DISCUSSION THREAD

**Discussion by SilencedPerson** [godotengine/godot-proposals#4202] [Scripting]
The question is in the title but here is how Javascript does it and i find it very useful.
![image](https://user-images.githubusercontent.com/82024211/157679460-6b49f42e-ed02-43c5-afaf-8a18d01289ac.png)

**Comment by Mickeon** [comment]
I personally find very valuable to have these functions available to the global scope. It makes the code more readable and less repetitive. It's probably why.

... Also, although it's been worked on, built-in static functions are not available, which means that for this to work "Math" would have to be a Singleton, which is... technically not necessary(?), nor exactly pretty and accessible to everyone, for the aforementioned reasons above.

**Reply by SilencedPerson** [reply]
Actually, just having them named as `math_*function name*()` would be enough for me I think.

**Reply by ghost** [reply]
I'd hate having any more length to the function names, myself, but I suppose that's a personal preference. [It probably doesn't help that I cut my teeth on the earliest versions of BASIC. :) ]

**Reply by Mickeon** [reply]
If it were for me I'd take notes on how **[RandomNumberGenerator](https://docs.godotengine.org/en/stable/classes/class_randomnumbergenerator.html)** is implemented for this. Because, as is, it's just nice to have plenty of general use math functions available in the global scope. For the future, however, if there's ever going to be the necessity to define more, advanced "Math" functions then by all means, it would be ideal to include them all in a hypothetical Math class, while still keeping the global scope intact.
